src/main/java/com/training/library/AddBookServlet.java

```diff
@@ -23,15 +22,13 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
 
     protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         PrintWriter writer = response.getWriter();
-
-        try {
-            databaseService = new DatabaseService();
-            Connection conn = databaseService.setConnection();
+        try(Connection conn = databaseService.setConnection()) {
```

Comment:
[here ](http://java-online.ru/jdbc-pool.xhtml)you can find how to create a connection pool